crates/polars-plan/src/plans/conversion/dsl_to_ir/expr_to_ir.rs

```diff
@@ -80,10 +47,42 @@ fn to_aexpr_impl_materialized_lit(
 }
 
 pub struct ExprToIRContext<'a> {
-    pub with_fields: Option<(Node, Schema)>,
+    pub(super) with_fields: Option<(Node, Schema)>,
     pub arena: &'a mut Arena<AExpr>,
     pub schema: &'a Schema,
+
     pub allow_unknown: bool,
+    /// Check whether mentioned column names exist in the schema.
```

Comment:
There is a small refactor here to make it such that can pass whether to check column names or not.